simple/stats/data.py

```diff
@@ -85,6 +100,49 @@ def triples(self) -> list[Triple]:
     ]
 
 
+@dataclass
+class Provenance:
+  id: str
+  source_id: str
+  name: str
+  url: str = ""
+
+  def triples(self) -> list[Triple]:
+    triples: list[Triple] = []
+    triples.extend([
+        Triple(self.id, _PREDICATE_TYPE_OF, object_id=_PROVENANCE),
+        Triple(self.id, _PREDICATE_NAME, object_value=self.name),
+        Triple(self.id, _PREDICATE_SOURCE, object_id=self.source_id),
+    ])
+    if self.url:
+      triples.append(Triple(self.id, _PREDICATE_URL, object_value=self.url))
+    return triples
+
+
+@dataclass
+class Source:
+  id: str
+  name: str
+  url: str = ""
+  domain: str = field(init=False)
+
+  def __post_init__(self):
+    self.domain = urlparse(self.url).netloc
```

Comment:
ok if self.url is ''?

Reply:
Yes. https://trinket.io/python3/5ce07302bf